### ISSUE TITLE
Creating Unique Index on VersionInfo table fails when Schema is changed

### ISSUE THREAD

**Issue by srbrill** [fluentmigrator/fluentmigrator#542]
I have setup a custom VersionTableMetaData class based off of DefaultVersionTableMetaData and changed the schema name, table name and the unique index name.  

When I ran my migration it fails when creating the index because the script created does no prepend the schema to the index or table.  Below is the error message

```
Ask For Help:
  http://groups.google.com/group/fluentmigrator-google-group
-------------------------------------------------------------------------------
[+] Using Database oracle and Connection String Data Source=APD;User Id=MyUser;Password=********;
SELECT 1 FROM ALL_USERS WHERE USERNAME = 'DifferentSchema'
SELECT 1 FROM ALL_TABLES WHERE upper(OWNER) = 'DifferentSchema' AND upper(TABLE_NAME) = 'DifferentVERSIONINFO'
SELECT 1 FROM ALL_TAB_COLUMNS WHERE upper(OWNER) = 'DifferentSchema' AND upper(TABLE_NAME) = 'DifferentVERSIONINFO' AND upper(COLUMN_NAME) = 'APPLIEDON'
-------------------------------------------------------------------------------
VersionUniqueMigration migrating
-------------------------------------------------------------------------------
[+] CreateIndex DifferentVERSIONINFO (Version)
CREATE UNIQUE INDEX UC_VersionDifferent ON DifferentVERSIONINFO (Version ASC)
!!! ORA-00942: table or view does not exist
!!! Oracle.DataAccess.Client.OracleException ORA-00942: table or view does not exist    at Oracle.DataAccess.Client.OracleException.HandleErrorHelper(Int32 errCode, OracleConnection conn, IntPtr opsErrCtx, OpoSqlValCtx* pOpoSqlValCtx, Object src, String procedure)
   at Oracle.DataAccess.Client.OracleException.HandleError(Int32 errCode, OracleConnection conn, String procedure, IntPtr opsErrCtx, OpoSqlValCtx* pOpoSqlValCtx, Object src)
   at Oracle.DataAccess.Client.OracleCommand.ExecuteNonQuery()
   at FluentMigrator.Runner.Processors.Oracle.OracleProcessor.Process(String sql) in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\Processors\Oracle\OracleProcessor.cs:line 210
   at FluentMigrator.Runner.Processors.ProcessorBase.Process(CreateIndexExpression expression) in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\Processors\ProcessorBase.cs:line 95
   at FluentMigrator.Expressions.CreateIndexExpression.ExecuteWith(IMigrationProcessor processor) in c:\Community\schambers\fluentmigrator\src\FluentMigrator\Expressions\CreateIndexExpression.cs:line 46
   at FluentMigrator.Runner.MigrationRunner.<>c__DisplayClass18.<ExecuteExpressions>b__15() in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\MigrationRunner.cs:line 402
   at FluentMigrator.Runner.StopWatch.Time(Action action) in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\StopWatch.cs:line 57
   at FluentMigrator.Runner.MigrationRunner.ExecuteExpressions(ICollection`1 expressions) in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\MigrationRunner.cs:line 415
   at FluentMigrator.Runner.MigrationRunner.ExecuteMigration(IMigration migration, Action`2 getExpressions) in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\MigrationRunner.cs:line 371
   at FluentMigrator.Runner.MigrationRunner.ApplyMigrationUp(IMigrationInfo migrationInfo, Boolean useTransaction) in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\MigrationRunner.cs:line 245
   at FluentMigrator.Runner.MigrationRunner.Up(IMigration migration) in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\MigrationRunner.cs:line 360
   at FluentMigrator.Runner.VersionLoader.LoadVersionInfo() in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\VersionLoader.cs:line 154
   at FluentMigrator.Runner.VersionLoader..ctor(IMigrationRunner runner, Assembly assembly, IMigrationConventions conventions) inc:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\VersionLoader.cs:line 43
   at FluentMigrator.Runner.MigrationRunner..ctor(Assembly assembly, IRunnerContext runnerContext, IMigrationProcessor processor) in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\MigrationRunner.cs:line 87
   at FluentMigrator.Runner.Initialization.TaskExecutor.Initialize() in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\Initialization\TaskExecutor.cs:line 53
   at FluentMigrator.Runner.Initialization.TaskExecutor.Execute() in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Runner\Initialization\TaskExecutor.cs:line 60
   at FluentMigrator.Console.MigratorConsole.ExecuteMigrations(IAnnouncer announcer) in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Console\MigratorConsole.cs:line 307
   at FluentMigrator.Console.MigratorConsole.ExecuteMigrations() in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Console\MigratorConsole.cs:line 254
   at FluentMigrator.Console.MigratorConsole..ctor(String[] args) in c:\Community\schambers\fluentmigrator\src\FluentMigrator.Console\MigratorConsole.cs:line 217

```

**Comment by cwalv** [comment]
I'm seeing this same problem when using `VersionTableMetaData` to override `SchemaName`.

```
/* Using Database OracleManaged and Connection String ********** */
SELECT 1 FROM ALL_USERS WHERE USERNAME = 'SIMS'
SELECT 1 FROM ALL_TABLES WHERE upper(OWNER) = 'SIMS' AND upper(TABLE_NAME) = 'VERSIONINFO'
/* PREVIEW-ONLY MODE ========================================================= */

/* VersionMigration migrating ================================================ */

/* CreateTable VersionInfo */
CREATE TABLE SIMS.VersionInfo (Version NUMBER(19,0) NOT NULL)

/* VersionMigration migrated */

SELECT 1 FROM ALL_TAB_COLUMNS WHERE upper(OWNER) = 'SIMS' AND upper(TABLE_NAME) = 'VERSIONINFO' AND upper(COLUMN_NAME) = 'APPLIEDON'
/* VersionUniqueMigration migrating ========================================== */

/* CreateIndex VersionInfo (Version) */
CREATE UNIQUE INDEX SIMS.IDX_VersionInfo_Version ON VersionInfo (Version ASC)

/* AlterTable VersionInfo */
/* No SQL statement executed. */

/* CreateColumn VersionInfo AppliedOn DateTime */
ALTER TABLE VersionInfo ADD AppliedOn TIMESTAMP(4)

/* VersionUniqueMigration migrated */

SELECT 1 FROM ALL_TAB_COLUMNS WHERE upper(OWNER) = 'SIMS' AND upper(TABLE_NAME) = 'VERSIONINFO' AND upper(COLUMN_NAME) = 'DESCRIPTION'
/* VersionDescriptionMigration migrating ===================================== */

/* AlterTable VersionInfo */
/* No SQL statement executed. */

/* CreateColumn VersionInfo Description String */
ALTER TABLE VersionInfo ADD Description NVARCHAR2(1024)
```

Fails because the `CREATE INDEX` and `ALTER TABLE` statements don't include the schema part of the table name.  Running this manually ahead of time works:

```
CREATE TABLE SIMS.VersionInfo (Version NUMBER(19,0) NOT NULL)
;
CREATE UNIQUE INDEX SIMS.IDX_VersionInfo_Version ON SIMS.VersionInfo (Version ASC)
;
/* CreateColumn VersionInfo AppliedOn DateTime */
ALTER TABLE SIMS.VersionInfo ADD AppliedOn TIMESTAMP(4)
;
/* CreateColumn VersionInfo Description String */
ALTER TABLE SIMS.VersionInfo ADD Description NVARCHAR2(1024)
```

**Comment by fubar-coder** [comment]
The schema handling of ORACLE (and other databses) was streamlined and this problem should be fixed in 2.0.0. Please leave a comment if the issue is still there after the 2.0 release.

**Comment by rodrigoramos** [comment]
This bug was partially fixed.

When creating an index it's targeting the table correctly although is not using the schema in the index name.

The output SQL is:
```
CREATE UNIQUE INDEX UC_Version ON <SchemaName>.VersionInfo (Version ASC)
```
Where it should be:

```
CREATE UNIQUE INDEX <SchemaName>.UC_Version ON <SchemaName>.VersionInfo (Version ASC)
```

I'm using version 3.2.1.
